build-support/bin/generate_github_workflows.py

```diff
@@ -162,36 +155,37 @@ def checkout(*, containerized: bool = False) -> Sequence[Step]:
                 "run": 'git config --global safe.directory "$GITHUB_WORKSPACE"',
             }
         )
-    steps.extend(
-        [
-            # For a push event, the commit we care about is HEAD itself.
-            # This CI currently only runs on PRs, so this is future-proofing.
-            {
-                "name": "Get commit message for branch builds",
-                "if": "github.event_name == 'push'",
-                "run": dedent(
-                    """\
-                echo "COMMIT_MESSAGE<<EOF" >> $GITHUB_ENV
-                echo "$(git log --format=%B -n 1 HEAD)" >> $GITHUB_ENV
-                echo "EOF" >> $GITHUB_ENV
-                """
-                ),
-            },
-            # For a pull_request event, the commit we care about is the second parent of the merge
-            # commit. This CI currently only runs on PRs, so this is future-proofing.
-            {
-                "name": "Get commit message for PR builds",
-                "if": "github.event_name == 'pull_request'",
-                "run": dedent(
-                    """\
-                echo "COMMIT_MESSAGE<<EOF" >> $GITHUB_ENV
-                echo "$(git log --format=%B -n 1 HEAD^2)" >> $GITHUB_ENV
-                echo "EOF" >> $GITHUB_ENV
-                """
-                ),
-            },
-        ]
-    )
+    if get_commit_msg:
```

Comment:
The job now requires us to do the checkout, but it doesn't care about the commit message, so we allow fetching it to be skipped.